src/provision/freedom-module.js

```diff
@@ -0,0 +1,351 @@
+//var XMLHttpRequest = require('freedom-xhr').corexhr;
+// Included directly in freedom module. node-forge not browserify-friendly
+//var forge = require('node-forge')({ disableNativeCode: true });
+
+var POLL_TIMEOUT = 5000; //milliseconds
+
+var STATUS_CODES = {
+  "START": "Starting provisioner",
+  "OAUTH_INIT": "Initializing oauth flow",
+  "OAUTH_ERROR": "Error getting oauth token",
+  "OAUTH_COMPLETE": "Got oauth token",
+  "SSHKEY_RETRIEVED": "Retrieved SSH keys from storage",
+  "SSHKEY_GENERATED": "Generated new SSH keys",
+  "CLOUD_FAILED": "Failed to complete cloud operation",
+  "CLOUD_INIT_ADDKEY": "Starting to add SSH key to cloud account",
+  "CLOUD_DONE_ADDKEY": "Done adding SSH key to cloud account",
+  "CLOUD_INIT_VM": "Starting to provision VM",
+  "CLOUD_DONE_VM": "Done provisioning VMs",
+  "CLOUD_WAITING_VM": "Waiting on VM",
+};
+
+var REDIRECT_URIS = [
+  "https://kmnjcbmibpajfljogomikcdlmhgpnolg.chromiumapp.org"
```

Comment:
https://developer.chrome.com/apps/identity

Right now the same app will use the built-in core.oauth provider in freedom-for-chrome, which uses chrome.identity. This is the format of redirect URIs that need to be registered in order for chrome.identity to properly get the token.